src/pgo/model/distsys/StateStrategy.java

```diff
@@ -0,0 +1,22 @@
+package pgo.model.distsys;
+
+import pgo.model.golang.Expression;
+import pgo.model.golang.GoProgram;
+import pgo.model.golang.Statement;
+import pgo.model.intermediate.PGoVariable;
+
+import java.util.Vector;
+import java.util.stream.Stream;
+
+// TODO flesh out this interface
+public interface StateStrategy {
+	void generateConfig(GoProgram go);
+	void generateGlobalVariables(GoProgram go);
+	void initializeGlobalVariables(GoProgram go);
+	void lock(int lockGroup, Vector<Statement> stmts, Stream<PGoVariable> vars);
+	void unlock(int lockGroup, Vector<Statement> stmts, Stream<PGoVariable> vars);
+
+	// FIXME flesh these out
```

Comment:
What exactly do you think is inappropriate? Would be interesting to discuss here.

Reply:
this is exposing an API to get and set whole variables, e.g. whole lists, whole sets. This might not be what we want in some cases, e.g. setting a single element of a list.

Reply:
Right, that makes sense. I think for now we can stick with the interface you proposed and change it as we add support for more fine-grained variable updates.